harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -582,11 +582,21 @@ def _load(self) -> None:
             ["checkpoint.pt"],
         ]
 
+        checkpoint: Optional[Dict[str, Any]] = None
         for ckpt_path in potential_paths:
             maybe_ckpt = self.load_path.joinpath(*ckpt_path)
             if maybe_ckpt.exists():
                 checkpoint = torch.load(str(maybe_ckpt), map_location="cpu")  # type: ignore
                 break
+        if checkpoint is None or not isinstance(checkpoint, dict):
+            # CHECK me
+            return
+
+        for callback in self.callbacks.values():
+            # QUESTION: should we encourage users to return new vals instead of
```

Comment:
LightningModule used the original reference so that you could modify the checkpoint in place. But it looks like either way works.